enterprise_access/apps/api_client/license_manager_client.py

```diff
@@ -104,6 +110,90 @@ def assign_licenses(self, user_emails, subscription_uuid):
             logger.exception(exc)
             raise
 
+    def get_customer_agreement(self, customer_uuid):
+        """
+        Fetches the first customer agreement for the customer with the given uuid,
+        returns None if no such agreements exist. The resulting dictionary also contains
+        a list of all active subscription plans for the agreement in the "subscriptions" key.
+        """
+        endpoint = self.customer_agreement_endpoint + f'?enterprise_customer_uuid={customer_uuid}'
+        response = self.client.get(endpoint)
+
+        try:
+            response.raise_for_status()
+            results = response.json().get('results', [])
+            if not results:
+                return None
+            return results[0]
+        except requests.exceptions.HTTPError as exc:
+            logger.exception(
+                'Error fetching customer agreements for customer id %s, response: %s, exc: %s',
+                customer_uuid, response.content.decode(), exc,
+            )
```

Comment:
I know it's totally a drop in the bucket to just start doing this now, but if we format our human-readable key-value pairs using `key=value` syntax, then the datadog grok parser will automatically extract them into log attributes:
![Screenshot 2025-04-15 at 7 01 43 PM](https://github.com/user-attachments/assets/1f7bb3e4-c235-49dd-89e5-374904e44e2d)